HDSAviz/data_processing.py

```diff
@@ -0,0 +1,218 @@
+                If you don't specify this parameter all rows
+                will be read.
+    drop      : A list of strings indicating which (if any)
+                of the named columns you do not want to include
+                in the resulting dataframe. (ex. ['cats', 'dogs'],
+                default is not to drop any rows)
+    sep       : String indicating the column separator in the
+                file (optional, default = ',').
+
+    Returns:
+    --------
+    df : A pandas dataframe with the contents of the file,
+         limited to the number of rows specified and without the
+         columns named in "drop".
+    """
+    df = pd.read_csv(filename, sep=sep, nrows=numrows)
+    if not drop:
+        df.drop(drop, axis=1, inplace=True)
+
+    return df
+
+
+def get_params(numrows=None, drop=['End_time', 'Oxygen']):
+    """
+    NOTE: This function is specific to our lignin modeling dataset
+
+    Returns a pandas dataframe with all the parameters analyzed in
+    the sensitivity analysis, but not additional parameters like
+    end time and oxygen content.  If you would like all of the
+    parameters (even those not analyzed for sensitivity) then pass
+    drop=False.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which parameters you do not want to
+              include in the returned dataframe.  If you want all params
+              then pass drop=False.
+
+    Returns:
+    -------
+    A pandas dataframe
+
+    """
+    return read_file('../../HDSAviz_data/input_parameters.csv',
+                     numrows=numrows, drop=drop)
+
+
+def get_results(numrows=None, drop=['light_aromatic_C-C',
+                                    'light_aromatic_methoxyl']):
+    """
+    NOTE: This function is specific to our lignin modeling dataset
+
+    Returns a pandas dataframe with the results of running all of the
+    simulations for the parameters sets in `input_parameters.csv`. This
+    function drops two unused functional groups from the results file.
+
+    Parameters:
+    ----------
+    numrows : the number of rows of the input_parameters file to read
+              (default is to read all rows).
+    drop    : a list of strings for which output measures to drop from
+              the returned dataframe.  If you want all outputs use
+              drop=False.
+    """
+    return read_file('../../HDSAviz_data/results.csv',
+                     numrows=numrows, drop=drop)
+
+
+def get_sa(path='../../HDSAviz_data/'):
+    """
+    This function reads and processes all the sensitivity analysis results
+    in a specified folder, and returns a dictionary with the corresponding
+    dataframes.  Sensitivity analysis results should be in the default SALib
+    format and must start with the letter 'a'.
+
+    NOTE: there are two lines of code at the beginning of this function
```

Comment:
I think this note shall go at the beginning of the file as well as in readme for this file.